Refresher/UI/PipelineForm.cs

```diff
@@ -167,13 +204,98 @@ private void OnButtonClick(object? sender, EventArgs e)
+        
+        if (this._pipeline.State == PipelineState.Running)
+            return;
+        
+        this.AddFormInputsToPipeline();
+        
+        Task.Run(async () =>
+        {
+            try
+            {
+                List<GameInformation> games = await this._pipeline!.DownloadGameListAsync();
+                await Application.Instance.InvokeAsync(() =>
+                {
+                    this.HandleGameList(games);
+                });
+            }
+            catch (Exception ex)
+            {
+                State.Logger.LogError(LogType.Pipeline, $"Error while downloading games list: {ex}");
+            }
+        });
+    }
+
+    private void HandleGameList(List<GameInformation> games)
+    {
+        if (this._connectButton != null)
+        {
+            this._connectButton.Enabled = false;
+            this._connectButton.Text = "Connected!";
+        }
+        
+        foreach (GameInformation game in games)
+        {
+            GameItem item = new()
+            {
+                Text = $"{game.Name} [{game.TitleId} {game.Version}]",
+                Version = game.Version ?? "00.00",
+                TitleId = game.TitleId,
+            };
+
+            if (GameCacheAccessor.IconExistsInCache(game.TitleId))
+            {
+                using Stream iconStream = GameCacheAccessor.GetIconFromCache(game.TitleId);
+                try
+                {
+                    item.Image = new Bitmap(iconStream).WithSize(new Size(64, 64));
+                }
+                catch (NotSupportedException)
+                {
+                    // Failed to set image for NPEB01899: System.NotSupportedException: No imaging component suitable to complete this operation was found.
+                    // ignore for now
+                }
+                catch (FormatException)
+                {
+                    // Failed to set image for BLUS31426: System.IO.FileFormatException: The image format is unrecognized.
+                    // also ignore for now
+
+                    // FileFormatException seems to not exist, but this page mentions it extending FormatException:
+                    // https://learn.microsoft.com/en-us/dotnet/api/system.io.fileformatexception
+                }
+                catch(Exception e)
+                {
+                    State.Logger.LogWarning(InfoRetrieval, $"Failed to set image for {game}: {e}");
+                    SentrySdk.CaptureException(e);
+                }
+            }
+
+            this._gamesDropDown!.Items.Add(item);
+        }
+
+        this._gamesDropDown!.Enabled = true;
+    }
+
     private void OnAutoDiscoverClick(object? sender, EventArgs e)
     {
         if (this._pipeline == null)
             return;
         
         if (this._pipeline.State == PipelineState.Running)
             return;
+
+        if (this._autoDiscoverCts != null)
+        {
+            this._autoDiscoverCts.Cancel();
+            return;
+        }
```

Comment:
Maybe i'm misunderstanding how these work, but is this ever set back to `null` after its cancelled?

Reply:
It gets set to null after the try-catch in the Task.